elastic-search-aggregator/src/main/java/in/ramanujam/elasticsearch/aggregator/ElasticSearchAggregatorScheduledTask.java

```diff
@@ -12,19 +12,20 @@
 @Component
 public class ElasticSearchAggregatorScheduledTask
 {
-    // TODO: how can we stop it from running after all records are persisted?
     @Scheduled(fixedDelay = 100)
     public void runWithDelay() throws IOException
     {
         List<MinerRecord> records = ElasticSearchAggregator.getInstance().retrieveAllRecords();
         for( MinerRecord esRecord : records )
         {
-            ElasticSearchAggregator.getInstance().
-                    moveRecordFromElasticSearchToMongo( esRecord, MongoUtils.getCollection() );
+            if( !esRecord.equals( new MinerRecord()) )
+            {
+                ElasticSearchAggregator.getInstance().
+                        moveRecordFromElasticSearchToMongo( esRecord, MongoUtils.getCollection() );
+            }
         }
-            // TODO: replace these conditions:
-        if( records.size() == 1 && records.get( 0 ).equals( new MinerRecord(  ) )
-                && ElasticSearchAggregator.getInstance().isElasticSearchFillerFinished() ) // TODO: это фикс для бага - при записи в ES создается еще одна запись со всеми параметрами null
+
+        if( ElasticSearchAggregator.getInstance().isElasticSearchFillerFinished() && records.size() == 1 )
```

Comment:
what does it mean `records.size() == 1`? Maybe move to another method or add comment